src/compiler/checker.ts

```diff
@@ -4499,7 +4525,19 @@ namespace ts {
                         if (source === numberType && target.flags & TypeFlags.Enum) return Ternary.True;
                     }
                 }
+
+                if (relation === assignableRelation && source.flags & TypeFlags.ObjectLiteral && source.flags & TypeFlags.FreshObjectLiteral) {

```

Comment:
It seems strange that we only do this for assignability. Prior to this, every pair of types that return true for subtype also return true for assignability. That is no longer true, and in fact, you will observe this in overload resolution.

```ts
interface MyObject {
   prop1: string;
}
declare function foo(param: MyObject): any;

foo({ prop1: "", prop2: "" }); // fails
```
This will fail, but if you add a second overload, it suddenly succeeds:
```ts
declare function foo(param: MyObject): any;
declare function foo(param: string): any;
```